### PR TITLE
Organization search fixes #44

### LINKED ISSUE #44: create organization search page

### PR DESCRIPTION
**Pull request by jsheng112**
add basic search for discovering new organizations and saving those organizations. Organizations are searched by prefix.
I accidentally created this branch in the same branch as the last PR, but I will keep this PR very simple and try to cherry pick those commits once merged to create a linear history.

### CODE REVIEW

**Comment by meltsufin** [comment]
It kind of sucks, but it's going to be tricky to review this PR before #31 is merged.
So, I would hold off reviewing this one for now.

File: capstone/src/main/java/com/step902020/capstone/OrganizationController.java [hunk 1]
```diff
@@ -66,4 +72,16 @@ public RedirectView saveOrganization(
     this.organizationRepository.save(current);
     return new RedirectView("profile.html", true);
   }
+
+  @GetMapping("search-organization")
+  public List<Organization> searchOrganization(
+      @RequestParam("name") String name, 
+      @RequestParam("university") String university) throws IOException {
+    
+    if (name.equals("")) {
```
Comment (lbourret) [inline capstone/src/main/java/com/step902020/capstone/OrganizationController.java:81]: I like that it returns all if it's an empty search

File: capstone/src/main/resources/static/organizationsearch.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <meta>
+    <title>My Account</title>
+    <link rel="stylesheet" href="style.css">
+    <script src="script.js"></script>
+</head>
+<body onload="searchOrg();">
+    <nav class="nav-bar">
+        <ul>
+            <li class="home"><a href="index.html">College Activity Finder</a></li>
+            <li><a href="">Events</a></li>
```
Comment (lbourret) [inline capstone/src/main/resources/static/organizationsearch.html:13]: Maybe we should switch index.html to under events and have the College activity finder act like a logo

File: capstone/src/main/resources/static/organizationsearch.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <br/>
+    
+    
+    <input type="text" placeholder="Search by organization name" id="search-org">
+    <button class="button submit" onclick="searchOrg();">Search</button>
+    <div id="no-profile">
+        <p>You have not built your profile yet, please submit a profile </p>
+    </div>
+    <div id="main">
+        <div id="list-organizations"></div>
```
Comment (lbourret) [inline capstone/src/main/resources/static/organizationsearch.html:28]: really efficient!

File: capstone/src/main/resources/static/organizationsearch.html (hunk 2 above)
Comment (oasemota99) [reply capstone/src/main/resources/static/organizationsearch.html:13]: A lot of websites will use their logo as a way to get back to the home page, but I think either choice is fine

Comment (jsheng112) [reply capstone/src/main/resources/static/organizationsearch.html:13]: What about putting an index link for both?

File: capstone/pom.xml [hunk 4]
```diff
@@ -84,7 +84,6 @@
         <version>2.2.0</version>
         <configuration>
           <deploy.projectId>step90-2020</deploy.projectId>
-          <deploy.version>1</deploy.version>
```
[the remark this answers is not in the document]
Comment (jsheng112) [reply capstone/pom.xml:87]: I will wait for him to merge first.

[the remark this answers is not in the document]
Comment (meltsufin) [reply capstone/pom.xml:87]: Merged. Btw, the rule we have on our team is that if a PR is approved and the original author is not around to merge it in a reasonable amount of time, anyone can merge it. :)

File: capstone/src/main/java/com/step902020/capstone/OrganizationRepository.java [hunk 5]
```diff
@@ -10,4 +12,9 @@
   public long deleteByEmail(String email);
 
   public List<Organization> findAllByEmail(List<String> email);
+
+  public List<Organization> findByUniversity(String university);
+
+  @Query("select * from organization where name >= @name and name < @endname and university = @university")
```
Comment (meltsufin) [inline capstone/src/main/java/com/step902020/capstone/OrganizationRepository.java:18]: Neat!

File: capstone/src/main/resources/static/script.js [hunk 6]
```diff
@@ -416,3 +441,23 @@ function closeForm() {
...
+    getUserType();
+    if (sessionStorage.getItem("university") == null) {
+      return;
+    }
+  }
+  var name = document.getElementById("search-org").value;
+  fetch('search-organization?name=' + name + "&university=" + university).then(response => response.json()).then((organizations) => {
+
+    const orgListElement = document.getElementById('list-organizations');
+    orgListElement.innerHTML = '';
```
Comment (meltsufin) [inline capstone/src/main/resources/static/script.js:457]: What if 0 results are returned? Should the page say something like "No organizations found"? 
Doesn't need to be done in this PR.

Comment (jsheng112) [reply capstone/src/main/resources/static/script.js:457]: I will file an issue to address it in a later PR.

**Pull request merged**